pyproject.toml

```diff
@@ -31,7 +31,9 @@ dependencies = [
     "defusedxml >=0.7.1,<1.0.0",
     "packageurl-python >= 0.11.1,<1.0.0",
     "ruamel.yaml >= 0.18.6,<1.0.0",
-    "jsonschema >= 4.22.0,<5.0.0"
+    "jsonschema >= 4.22.0,<5.0.0",
+    "beautifulsoup4==4.12.3",
+    "types-beautifulsoup4==4.12.0.20240511"
```

Comment:
We prefer setting a range for the dependencies (as the dependencies above) instead of pinning the exact version. This is because we want to let pip handle the dependency resolution and find the optimal version for each dependency. Forcing it to use a specific version might cause conflicts between the packages when you run `pip install`. 
In this case, you can specify it something like     
```
"beautifulsoup4 >= 4.12.0,<5.0.0",
"types-beautifulsoup4 >= 4.12.0,<5.0.0"
```
In additions, please move the entry for `types-beautifulsoup4` to the `[project.optional-dependencies].dev` section, where you also find other "types" packages defined there.

Reply:
Modified

Reply:
Thanks. Even though the range issue is fixed, `types-beautifulsoup4` entry hasn't been moved to `[project.optional-dependencies].dev` in `pyproject.toml`, just to let you know :smile:

Reply:
Should I add both or just the `types-beautifulsoup4`?

Reply:
Oh I meant you keep `beautifulsoup4` right where it currently is an **only** move `types-beautifulsoup4` to `[project.optional-dependencies].dev` in `pyproject.toml`. This is so that   `types-beautifulsoup4`  is only installed in the dev environment, and not in the final Macaron artifact that is installed within the Docker image.